static/js/components/EmploymentTab.js

```diff
@@ -65,56 +79,232 @@ class EmploymentTab extends ProfileTab {
+  }
+
+  closeWorkDialog = () => {
+    const { setWorkDialogVisibility } = this.props;
+    setWorkDialogVisibility(false);
+  }
+
+  addWorkHistoryEntry = () => {
+    const {
+      updateProfile,
+      profile,
+      setWorkDialogIndex,
+      setWorkDialogVisibility,
+    } = this.props;
+    let clone = Object.assign({}, profile);
+    clone['work_history'] = clone['work_history'].concat(
+      _.cloneDeep(this.blankWorkHistoryEntry)
+    );
+    updateProfile(clone);
+    setWorkDialogIndex(clone.work_history.length - 1);
+    setWorkDialogVisibility(true);
+  }
+
+  deleteWorkHistoryEntry = index => {
+    const { updateProfile, profile } = this.props;
+    let clone = _.cloneDeep(profile);
+    clone['work_history'].splice(index, 1);
+    updateProfile(clone);
+  }
+
+  editWorkHistoryForm () {
+    const { ui } = this.props;
+    let keySet = (key) => ['work_history', ui.workDialogIndex, key];
     return (
-      <Grid key={index} className="profile-tab-grid">
+      <Grid className="profile-tab-grid no-border">
+        <Cell col={12} className="employment-form-title">
+          Add Employment
+        </Cell>
         <Cell col={12}>
           {this.boundTextField(keySet('company_name'), 'Company Name')}
         </Cell>
-        <Cell col={12}>
-          {this.boundTextField(keySet('position'), 'Position')}
+        <Cell col={4}>
+          {this.boundSelectField(keySet('country'), 'Country', this.countryOptions)}
         </Cell>
-        <Cell col={12}>
-          {this.boundTextField(keySet('industry'), 'Industry')}
+        <Cell col={4}>
+          {this.boundStateSelectField(keySet('state_or_territory'), keySet('country'), 'State or Territory')}
         </Cell>
-        <Cell col={12}>
+        <Cell col={4}>
           {this.boundTextField(keySet('city'), 'City')}
         </Cell>
         <Cell col={12}>
-          {this.boundStateSelectField(keySet('state_or_territory'), keySet('country'), 'State or Territory')}
+          {this.boundTextField(keySet('industry'), 'Industry')}
         </Cell>
         <Cell col={12}>
-          {this.boundSelectField(keySet('country'), 'Country', this.countryOptions)}
+          {this.boundTextField(keySet('position'), 'Position')}
         </Cell>
-        <Cell col={12}>
+        <Cell col={6}>
           {this.boundMonthYearField(keySet('start_date'), 'Start Date')}
         </Cell>
-        <Cell col={12}>
+        <Cell col={6}>
           {this.boundMonthYearField(keySet('end_date'), 'End Date')}
+          <span className="end-date-hint">
+            Leave blank if this is a current position
+          </span>
+        </Cell>
+      </Grid>
+    );
+  }
+
+  renderWorkHistory () {
+    const { ui, profile: { work_history } } = this.props;
+    if ( _.isUndefined(ui.workHistoryEdit) || ui.workHistoryEdit === true ) {
+      let workHistory = [];
+      if ( !_.isEmpty(work_history) ) {
+        workHistory.push(...work_history.map((entry, i) => this.jobRow(entry, i)));
```

Comment:
I think the current state is an artifact from an refactor - I was declaring `workHistory`, adding some stuff to it, and then pushing the `jobRows` on. I didn't change things around when I moved stuff around I guess! haha